sky/skylet/providers/azure/node_provider.py

```diff
@@ -65,12 +65,8 @@ def __init__(self, provider_config, cluster_name):
         _configure_resource_group({"provider": provider_config})
         subscription_id = provider_config["subscription_id"]
         self.cache_stopped_nodes = provider_config.get("cache_stopped_nodes", True)
-        # AWS provides managed identity for Azure, but it is not setup properly by
-        # default. This interferes with azure-cli credentials and causes failures,
-        # when using sky to launch Azure on AWS ec2 instances. We disable it to give
-        # way to azure-cli credentials.
-        credential = DefaultAzureCredential(exclude_shared_token_cache_credential=True,
-                                            exclude_managed_identity_credential=True)
+        # Sky only supports Azure CLI credential for now.
```

Comment:
cc @suquark  to take a look.

IIUC, @suquark added the change here.  According to https://github.com/skypilot-org/skypilot/issues/1332#issuecomment-1296139810 if this new change doesn't fix the non-deterministic error maybe we should keep the old version?

Reply:
Since we asked the user to generate the credentials with `azure-cli` in our documents and our`sky check` only checks the azure-cli crendentials, we may want to keep it consistent here for the credential to avoid unexpected errors caused by another credential manager, e.g. the `exclude_managed_identity_credential` we added before was to avoid the azure credential manager confusing the `node_provider`.

Reply:
I am ok with this change, since we really only need azure cli credential. allowing other credentials may result in using wrong credentials. we can revert it later if really other credentials are required later